reference/day_procedures.py

```diff
@@ -0,0 +1,158 @@
+"""Generate a list of procedures that may be performed as day cases or outpatients"""
+
+import json
+import sys
+
+import pyspark.sql.functions as F
+from databricks.connect import DatabricksSession
+from pyspark.sql import DataFrame, SparkSession
+from scipy.stats import binomtest
+
+from nhp_datasets.apc import hes_apc
+from nhp_datasets.providers import read_data_with_provider
+
+
+def _get_procedures(spark: SparkSession, table_name: str) -> DataFrame:
+    return (
+        spark.read.table(f"hes.silver.{table_name}_procedures")
+        .filter(F.col("procedure_order") == 1)
+        .filter(~F.col("procedure_code").rlike("^O(1[1-46]|28|3[01346]|4[2-8]|5[23])"))
+        .filter(~F.col("procedure_code").rlike("^X[6-9]"))
+        .filter(~F.col("procedure_code").rlike("^[UYZ]"))
+    )
+
+
+def get_day_procedure_code_list(
+    spark: SparkSession, minimum_total: int = 100, p_value: float = 0.001
+) -> dict[str, list[str]]:
+    """_summary_
+
+    :param spark: The spark session to use
+    :type spark: SparkSession
+    :param minimum_total: what is the minimum number of procedures in total that must be performed,
+        defaults to 100
+    :type minimum_total: int, optional
+    :param p_value: what p-value to use when we run the binomial tests, defaults to 0.001
+    :type p_value: float, optional
+    :return: a dictionary containing the types and their code lists
+    :rtype: dict[str, list[str]]
+    """
+    providers = (
+        spark.read.table("strategyunit.reference.ods_trusts")
+        .filter(F.col("org_type").startswith("ACUTE"))
+        .select(F.col("org_to").alias("provider"))
+        .distinct()
+        .persist()
+    )
+
+    fyear_criteria = F.col("fyear") == 201920
```

Comment:
Why are we using 201920?

Reply:
because I wrote this code a long time ago! 🤣